src/User32/PublicAPI.Unshipped.txt

```diff
@@ -34,14 +34,6 @@ PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS.DDC_DEFAULT = 0 -> PInvoke.User32.DIA
 PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS.DDC_DISABLE_ALL = 1 -> PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS
 PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS.DDC_DISABLE_CONTROL_RELAYOUT = PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS.DDC_DISABLE_ALL | PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS.DDC_DISABLE_RESIZE -> PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS
 PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS.DDC_DISABLE_RESIZE = 2 -> PInvoke.User32.DIALOG_DPI_CHANGE_BEHAVIORS
-PInvoke.User32.DISPLAY_DEVICE
```

Comment:
**Typical advice here would be**
To avoid a binary breaking change, in the place of this type you must add `[assembly: TypeForwardedTo(typeof(DISPLAY_DEVICE))]`. Please place these forwarders in the original source files where the code used to be.
Same for any other type(s) you move.

**However**
Since you're also stripping the moved types of their nesting class, it can't be a binary compatible change. We should take this if/when we're willing to take a binary breaking change. It's probably the right change to make, but we should be thoughtful about how to roll it out.

Reply:
I was relying on the fact that this change hasn't been published.

Reply:
I was relying on the fact that this change hasn't been published, I thought it won't be a hassle.
Anyway I'll add these attributes.
So just to be clear, you want me to keep the `DISPLAY_DEVICE` in both places but mark it as `TypeForwardedTo` in the old one?

Reply:
I was relying on the fact that this change hasn't been published, I thought it won't be a hassle.

Anyway, I'm working on the forwarding, so we're going to keep the `DISPLAY_DEVICE` in both places but decorate the old types with `TypeForwardedTo`?
What about the method signature, what type does it specify in the parameters, the old or new one?

Reply:
https://docs.microsoft.com/en-us/cpp/extensions/type-forwarding-cpp-cli

> You cannot forward the following types:
> Generic types
> Native types
> Nested types (if you want to forward a nested type, you should forward the enclosing type)

An alternative might be to redefine these types in Core, create new methods, type etc definitions based on the new definition in Core, and mark the current definition, and all methods and types that rely on it, as Obsolete.

Reply:
So the new method should be an overload of the old one?

Reply:
> I was relying on the fact that this change hasn't been published,

I totally missed this. If these types are new, moving them is no problem.

Reply:
In User32 I added a new method called the same but different by the overload (i.e. `PInvoke.DEVICE_MODE` instead of `PInvoke.User32.DEVICE_MODE`) and it seems there is a problem.

I either get

_error CS0111: Type 'User32' already defines a member called 'EnumDisplayDevices' with the same parameter types_, or RS0016. Both end up being an IntPtr in this set:

```c#
static PInvoke.User32.EnumDisplayDevices(string lpDevice, uint iDevNum, System.IntPtr lpDisplayDevice, PInvoke.User32.EnumDisplayDevicesFlags dwFlags) -> bool
static PInvoke.User32.EnumDisplayDevices(string lpDevice, uint iDevNum, ref PInvoke.DISPLAY_DEVICE lpDisplayDevice, PInvoke.User32.EnumDisplayDevicesFlags dwFlags) -> bool
```

Reply:
> I totally missed this. If these types are new, moving them is no problem.

😂 ok

Reply:
@AArnott 
> I totally missed this. If these types are new, moving them is no problem.

Great, done.